src/device/device_cold.hpp

```diff
@@ -0,0 +1,53 @@
+//
+// Created by Dusan Klinec on 10/08/2018.
+//
+
+#ifndef MONERO_DEVICE_COLD_H
+#define MONERO_DEVICE_COLD_H
+
+#include "wallet/wallet2.h"
+#include <boost/function.hpp>
+
+
+namespace hw {
+
+  typedef struct wallet_shim {
+    boost::function<crypto::public_key (const tools::wallet2::transfer_details &td)> get_tx_pub_key_from_received_outs;
+  } wallet_shim;
+
+  class tx_aux_data {
+  public:
+    std::vector<std::string> tx_device_aux;  // device generated aux data
+    std::vector<cryptonote::address_parse_info> tx_recipients;  // as entered by user
+  };
+
+  class device_cold {
+  public:
+
+    using exported_key_image = std::vector<std::pair<crypto::key_image, crypto::signature>>;
+
+    /**
+     * Key image sync with the cold protocol.
+     * @param wallet
+     * @param transfers
+     * @param ski
```

Comment:
Others might disagree, but those are pointless as tautological and just take up space. Let's see if there are other opinions :)

Reply:
I agree, its auto-generated by IDE, I will remove it